client.go

```diff
@@ -587,6 +587,12 @@ func (sc *SnowthClient) do(ctx context.Context, node *SnowthNode,
 	}
 
 	r.Close = true
+	if headers != nil {
```

Comment:
The range operator on a map will skip the for loop if the map is nil.  So, the outer nil check shouldn't be needed.

Reply:
TIL, Fixed.